structural-variantcalling.wdl

```diff
@@ -131,86 +131,100 @@ workflow SVcalling {
+                    inputFile = sort.outputVcf,
+                    outputPath = prefix + '.changed_id.vcf',
+                    newId = "'${pair.right}\\_%CHROM\\_%POS\\_%END'"
             }
-        }
 
-        File toBeMergedVcfs = select_first([removeFpDupDel.outputVcf,removeMisHomRR.outputVcf, annotateDH.outputVcf])
-    }
+            if (runDupHold) {
+                call duphold.Duphold as annotateDH {
+                    input:
+                        dockerImage = dockerImages["duphold"],
+                        inputVcf = setId.outputVcf,
+                        bamFile = bamFile,
+                        bamIndex = bamIndex,
+                        referenceFasta = referenceFasta,
+                        referenceFastaFai = referenceFastaFai,
+                        outputPath = prefix + '.duphold.vcf',
+                        sample = pair.right
+                }
+
+                call bcftools.View as removeFpDupDel {
+                    input:
+                        dockerImage = dockerImages["bcftools"],
+                        inputFile = annotateDH.outputVcf,
+                        outputPath = prefix + '.duphold_filtered.vcf',
+                        include = "'(SVTYPE = \"DEL\" & FMT/DHFFC[0] < 0.7) | (SVTYPE = \"DUP\" & FMT/DHBFC[0] > 1.3)'"
+                }
+            }
 
-    call survivor.Merge as survivor {
-        input:
-            dockerImage = dockerImages["survivor"],
-            filePaths = toBeMergedVcfs,
-            outputPath = SVdir + 'survivor/' + sample + '.merged.vcf'
+            if (excludeMisHomRef) {
+                call bcftools.View as removeMisHomRR {
+                    input:
+                    dockerImage = dockerImages["bcftools"],
+                    inputFile = select_first([setId.outputVcf, removeFpDupDel.outputVcf]),
+                    outputPath = prefix + '.GT_filtered.vcf',
+                    excludeUncalled = true,
+                    exclude = "'GT=\"RR\"'"
+                }
+            }
+
+            File toBeMergedVcfs = select_first([removeMisHomRR.outputVcf, removeFpDupDel.outputVcf, setId.outputVcf])
+         }
+
+        call survivor.Merge as survivor {
+            input:
+                dockerImage = dockerImages["survivor"],
+                filePaths = toBeMergedVcfs,
+                outputPath = SVdir + 'survivor/' + svtype + '.union.vcf',
+                suppVecs = 1
+        }
+
+        call bcftools.View as getIntersections {
+            input:
+            dockerImage = dockerImages["bcftools"],
+            inputFile = survivor.mergedVcf,
+            outputPath = SVdir + 'survivor/' + svtype + '.isec.vcf',
+            exclude = "'SUPP=\"1\"'"
+        }
     }
 
     output {
-        File cleverPredictions = clever.predictions
-        File cleverVcf = mateclever.matecleverVcf
-        File mantaVcf = manta.mantaVCF
-        File dellyBcf = delly.dellyBcf
-        File dellyVcf = delly2vcf.outputVcf
```

Comment:
It might still be nice to have these available as their own outputs, rather than just having them in the rawVcfs. For some downstream workflow the calls from a specific caller might be needed. eg. strelka can use output from manta

Reply:
Sure! I agree, I will put them back.